### PR TITLE
Remove Theia settings and point to Code image

### LINKED ISSUE #4486: Remove theia option in settings
### Problem to solve

Now that we're deprecating the old editor (Theia) (see https://github.com/gitpod-io/gitpod/issues/4459), we can safely remove the editor (Default IDE) preference in settings once the deprecation has been made.

⚠️ **Warning**: This is currently blocked until the deprecation is complete and will most probably be ready-for-development in the next milestone. /cc @akosyakov

### PR DESCRIPTION
**Pull request by laushinka**
This PR:
1. Removes the Theia option in Preferences
2. Points the Theia alias to the Code stable image.

/werft with-clean-slate-deployment

Fixes #4486

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **laushinka** after the PR has been reviewed.

Associated issue: *#4486*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gtsiolis** [comment]
Looking at this now! 👀

File: chart/templates/server-deployment.yaml
Comment (gtsiolis) [inline chart/templates/server-deployment.yaml:163]: **question**: Shall we open a follow up issue to remove this later?

File: components/dashboard/src/settings/Preferences.tsx
Comment (gtsiolis) [inline components/dashboard/src/settings/Preferences.tsx:55]: **praise**: Thanks for also removing this, @laushinka! 🎗️

**Review by gtsiolis — APPROVED** [review]
Thanks for making this change @laushinka! 🌟 

UX looks good for new users. Wondering how we could test this for existing users who previously had Theia as the selected editor. FWIW, I could test this also in staging with a couple of test users.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 6a35f28d85e3f12b98ccad750bd70901e97803db</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5511#pullrequestreview-745802991" title="Approved">gtsiolis</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **laushinka** after the PR has been reviewed.

Associated issue: *#4486*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by laushinka** [comment]
> UX looks good for new users. Wondering how we could test this for existing users who previously had Theia as the selected editor. FWIW, I could test this also in staging with a couple of test users.

I tested it yesterday by setting the DB for the default IDE for my user to 'theia' (in my preview env), and then opening up a workspace. I can show you quickly if you're able to access the staging DB.

File: chart/templates/server-deployment.yaml
Comment (laushinka) [reply chart/templates/server-deployment.yaml:163]: Yes, we should create a follow-up issue to clean up the code if we don't have it yet.

File: components/dashboard/src/settings/Preferences.tsx
Comment (JanKoehnlein) [inline components/dashboard/src/settings/Preferences.tsx:26]: This is the reason why you wouldn't find any user with "theia" as their `additionalData.ideSettings.defaultIde` in the DB: if Theia is selected we just delete the setting (default). We should leave the if statement in and replace `'theia'` with `'code'` instead.

**Review by JanKoehnlein — CHANGES_REQUESTED** [review]

File: components/dashboard/src/settings/Preferences.tsx
Comment (laushinka) [reply components/dashboard/src/settings/Preferences.tsx:26]: I queried whether the defaultIde is not 'code' or 'code-latest' but should've queried whether they were null! Did that just now and there is plenty. Thanks for catching it. Will fix and test again.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5511#pullrequestreview-745802991" title="Approved">gtsiolis</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **jankoehnlein** after the PR has been reviewed.

Associated issue: *#4486*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by laushinka — COMMENTED** [review]
When the user selects code we don't have to delete the settings actually because the default is already 'code'. We can see here https://github.com/gitpod-io/gitpod/blob/f66598c82c64558fe08c8beef1e2ebc3d4b79160/components/gitpod-db/src/typeorm/user-db-impl.ts#L81

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 4d1118cd041a1a6363a420607fac6f3f48231fba</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5511#pullrequestreview-745802991" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5511#issuecomment-912625356" title="LGTM">JanKoehnlein</a>*

Associated issue: *#4486*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [JanKoehnlein]
- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [JanKoehnlein]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by JanKoehnlein — APPROVED** [review]
LGTM

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5511#pullrequestreview-745802991" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5511#pullrequestreview-746244337" title="Approved">JanKoehnlein</a>*

Associated issue: *#4486*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [JanKoehnlein]
- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [JanKoehnlein]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gtsiolis** [comment]
IT IS DONE! ☄️

**Pull request merged**
